crossfire/__init__.py

```diff
@@ -51,16 +58,34 @@ def occurrences(
     )
 
 
+def flatten_df(row, column_name):
+    column_data = row[column_name]
+    return Series(
+        {f"{column_name}_{key}": value for key, value in column_data.items()}
+    )
+
+
 def flatten(data, nested_columns=None):
     nested_columns = set(nested_columns or NESTED_COLUMNS)
     if not nested_columns.issubset(NESTED_COLUMNS):
         raise NestedColumnError(nested_columns)
-    if not data:
+    if isinstance(data, DataFrame) and not data.empty:
+        keys = set(data.columns) & nested_columns
+        for key in keys:
+            data = pd_concat(
+                [
+                    data.drop(key, axis=1),
+                    data.apply(flatten_df, args=(key,), axis=1),
+                ],
```

Comment:
Not really important, but [no need to use a list if we're not mutating the list](https://cuducos.me/2018/06/13/tuplas-e-listas-no-python.html):

```suggestion
                (
                    data.drop(key, axis=1),
                    data.apply(flatten_df, args=(key,), axis=1),
                ),
```